File: MusicManagerAPI-client/src/main/java/com/ldz/music/manager/model/UserMusicStatusDTO.java
package com.ldz.music.manager.model;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

/**
 * Created by ldalzotto on 25/04/2017.
 */
public class UserMusicStatusDTO {

    @NotNull
    private String username;

    @Valid
    @Size(min = 0)
    private List<MusicTypeDTO> musicTypeDTO;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public List<MusicTypeDTO> getMusicTypeDTO() {
        return musicTypeDTO;
    }

    public void setMusicTypeDTO(List<MusicTypeDTO> musicTypeDTO) {
        this.musicTypeDTO = musicTypeDTO;
    }
}

File: MusicManagerAPI-client/src/scala/com/ldz/music/manager/model/UserMusicStatusDTO.scala
package com.ldz.music.manager.model

import javax.validation.Valid
import javax.validation.constraints.{NotNull, Size}
import java.util.List

import scala.annotation.meta.field
import scala.beans.BeanProperty

/**
  * Created by Loic on 20/06/2017.
  */
case class UserMusicStatusDTO(@(NotNull@field) @BeanProperty username: String,
                              @(Size@field)(min = 0) @Valid @BeanProperty musicTypeDTO: List[MusicTypeDTO]) {

  def this() {
    this(null, null)
  }

}
